src/main/resources/com/google/api/codegen/java/http_request.snip

```diff
@@ -0,0 +1,233 @@
+@extends "java/common.snip"
+@extends "java/method_sample.snip"
+
+@snippet generate(apiFile)
+  {@renderFileHeader(apiFile.fileHeader)}
+
+  @@Generated("by GAPIC")
+  @@BetaApi
+  {@requestClassDef(apiFile.request)}
+@end
+
+@private requestClassDef(request)
+  public final class {@request.typeName} implements Message {
```

Comment:
I actually had in mind using a single template for all message types (i.e. including request message types). This means that non-request message types will have some extra stuff they technically don't need (`populateMapMethod`), but I think that's fine.